src/test.js

```diff
@@ -74,8 +74,10 @@ Test.prototype = {
 		config.current = this;
 
 		if ( this.module.testEnvironment ) {
+			delete this.module.testEnvironment.before;
 			delete this.module.testEnvironment.beforeEach;
 			delete this.module.testEnvironment.afterEach;
+			delete this.module.testEnvironment.after;
```

Comment:
I don't think so. We have this delete statement only because we're introducing new properties to the modules' hook: `before` and `after`. Anyone previously extending that module with these keywords would get their tests affected. 

That same could happen when we moved `setup` and `teardown` to `beforeEach` and `afterEach`. Similarly we could mention the introduction of `QUnit.skip` to the API that could  interfere with any other use of custom made methods.